python/lsst/meas/algorithms/loadReferenceObjects.py

```diff
@@ -0,0 +1,252 @@
+    """!Return flux and flux error keys
+
+    @param[in] schema  reference catalog schema
+    @param[in] filterName  name of camera filter
+    @return a pair of keys:
+        flux key
+        flux error key, if present, else None
+    @throw RuntimeError if flux field not found
+    """
+    fluxField = getRefFluxField(schema, filterName)
+    fluxErrField = fluxField + "Sigma"
+    fluxKey = schema[fluxField].asKey()
+    try:
+        fluxErrKey = schema[fluxErrField].asKey()
+    except Exception:
+        fluxErrKey = None
+    return (fluxKey, fluxErrKey)
+
+class LoadReferenceObjectsConfig(pexConfig.Config):
+    pixelMargin = pexConfig.RangeField(
+        doc = "Padding to add to 4 all edges of the bounding box (pixels)",
+        dtype = int,
+        default = 50,
+        min = 0,
+    )
+    defaultFilter = pexConfig.Field(
+        doc = "Default reference catalog filter to use if filter not specified in exposure; " + \
+            "if blank then filter must be specified in exposure",
+        dtype = str,
+        default = "",
+    )
+    filterMap = pexConfig.DictField(
+        doc = "Mapping of camera filter name: reference catalog filter name; " + \
+            "each reference filter must exist",
+        keytype = str,
+        itemtype = str,
+        default = {},
+    )
+
+class LoadReferenceObjectsTask(pipeBase.Task):
+    """!Abstract base class to load objects from reference catalogs
+
+    @anchor LoadReferenceObjectsTask_
+
+    @section meas_algorithms_loadReferenceObjects_Contents Contents
+
+     - @ref meas_algorithms_loadReferenceObjects_Purpose
+     - @ref meas_algorithms_loadReferenceObjects_Initialize
+     - @ref meas_algorithms_loadReferenceObjects_IO
+     - @ref meas_algorithms_loadReferenceObjects_Schema
+     - @ref meas_algorithms_loadReferenceObjects_Config
+
+    @section meas_algorithms_loadReferenceObjects_Purpose  Description
+
+    Abstract base class for tasks that load objects from a reference catalog
+    in a particular region of the sky.
+
+    Implementations must subclass this class, override the loadObjectsInBBox method,
+    and will typically override the value of ConfigClass with a task-specific config class.
+
+    @section meas_algorithms_loadReferenceObjects_Initialize   Task initialisation
+
+    @copydoc \_\_init\_\_
+
+    @section meas_algorithms_loadReferenceObjects_IO       Invoking the Task
+
+    @copydoc loadObjectsInBBox
+
+    @section meas_algorithms_loadReferenceObjects_Schema       Schema of the reference object catalog
+
+    Reference object catalogs are instances of lsst.afw.table.SimpleCatalog with the following schema
+    (other fields may also be present):
+    - coord: position of star on sky (an lsst.afw.coord.IcrsCoord)
+    - centroid: position of star on an exposure, if relevant (an lsst.afw.Point2D)
+    - hasCentroid: is centroid usable?
+    - *referenceFilterName*_flux: brightness in the specified reference catalog filter: 10^(-0.4*mag)
```

Comment:
Given you've got code for Janskys elsewhere (not maggies), I think the `10^(-0.4*mag)` is a relic.